src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -5,10 +5,52 @@
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MINIMAL_AGE = 18;
+    private static final int MINIMAL_PASSWORD_LENGTH = 6;
+    private static final int MINIMAL_LOGIN_LENGTH = 6;
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        if (user.getLogin() == null) {
+            throw new RegistrationException("Login cannot be null");
+        }
+
+        if (user.getPassword() == null) {
+            throw new RegistrationException("Password cannot be null");
+        }
+
+        if (!isValidLogin(user)) {
+            throw new RegistrationException("Invalid login format "
+            + user.getLogin() + ". Min number of characters: " + MINIMAL_LOGIN_LENGTH);
+        }
+
+        if (!isValidPassword(user)) {
+            throw new RegistrationException("Invalid login format "
+                    + user.getLogin() + ". Min number of characters: " + MINIMAL_PASSWORD_LENGTH);
+        }
+        if (!isValidAge(user)) {
+            throw new RegistrationException("Invalid login format "
+                    + user.getLogin() + ". Min allowed age is " + MINIMAL_AGE);
+        }
```

Comment:
better to refactor like this and throw exceptions inside methods
```suggestion
       validateLogin(user.getLogin());
       validatePassword(user.getPassword());
       validateAge(user.getAge());
```